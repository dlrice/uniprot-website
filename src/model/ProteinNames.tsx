import React from 'react';
import NameView from './NameView';
import { InfoList } from 'franklin-sites';
import { ProteinNamesDefault } from './uniprotkb/sections/NamesAndTaxonomyConverter';

type ProteinNamesDataProps = {
  recommendedName?: string;
  shortNames?: string;
  alternativeNames?: string[];
};

const processShortNames = (
  shortNameList: ProteinNamesDefault['shortNames']
) => {
  if (!shortNameList) {
    return;
  }
  return shortNameList.reduce((r, d) => `, ${d.value}`, '');
};

export const ProteinNames: React.FC<ProteinNamesDataProps> = ({
  recommendedName,
  shortNames,
  alternativeNames,
}) => {
  const props = {
    name: recommendedName,
    shortNames,
    alternativeNames,
  };
  return <NameView {...props} />;
};

export const EntryProteinNames: React.FC<ProteinNamesDataProps> = ({
  recommendedName,
  shortNames,
  alternativeNames,
}) => {
  const infoData = [
    {
      title: 'Recommended name',
      content: recommendedName,
    },
    {
      title: 'Short names',
      content: shortNames,
    },
    {
      title: 'Alternative names',
      content: alternativeNames,
    },
  ];
  return <InfoList infoData={infoData} />;
};
