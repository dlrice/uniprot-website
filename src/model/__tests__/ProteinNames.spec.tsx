import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ProteinNames } from '../ProteinNames';

configure({ adapter: new Adapter() });

describe('ProteinNames', () => {
  test('should render protein_name', () => {
    const data = {
      proteinDescription: {
        recommendedName: {
          fullName: {
            value: 'My protein',
          },
        },
        submissionNames: [
          {
            fullName: {
              value: 'A submission',
            },
          },
        ],
        shortNames: [
          {
            value: 'My prot',
          },
        ],
        ecNumbers: [
          {
            value: 'ec1234',
          },
        ],
        alternativeNames: [
          {
            fullName: {
              value: 'Some protein',
            },
          },
        ],
      },
    };
    const wrapper = shallow(<ProteinNames data={data} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
