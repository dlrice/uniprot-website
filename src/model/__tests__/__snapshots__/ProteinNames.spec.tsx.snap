// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProteinNames should render protein_name 1`] = `"<NameView name=\\"rec full Name\\" shortNames={[undefined]} alternativeNames={{...}} />"`;
